lib/susedistribution.pm

```diff
@@ -724,7 +725,7 @@ sub activate_console {
         else {
             my $nr = console_nr($console);
             $self->hyperv_console_switch($console, $nr);
-            my @tags = ("tty$nr-selected", "text-logged-in-$user");
+            my @tags = ("tty$nr-selected", "text-logged-in-$user", "emergency-shell");
```

Comment:
As `activate_console` can be called a lot of times but I would expect `emergency-shell` only during bootup (am I wrong?) I would prefer the handling to be done only for boot handling, not everywhere where this is called and introducing the overhead of checking against the emergency shell as well every time a console is activated.

Reply:
You have a valid point. The structure of the code used in this case though, is a bit perplexed. Would you have any recommendation on how to achieve the same result without mixing activate_console?

Reply:
How about the following: 
- detect stall during boot up (e.g. bootloader for more than 500 sec)
- select_root_console 
- check if emergency-shell -> handle
- else post_fail_hook

Reply:
Sorry if I can't give you a clear "yes" or "no" answer. Could you please take a look at what https://github.com/os-autoinst/os-autoinst-distri-opensuse/blob/master/lib/bootbasetest.pm#L7 is trying to do? This is the post_fail_hook of the base module "bootbasetest". Every test module that handles "boot" can use that as base class and hence at any point in time if the job module fails the post_fail_hook should be able to detect if we at least reached a login prompt or emergency-shell, etc. Maybe you can try to extend that module instead?

And then there is https://github.com/os-autoinst/os-autoinst-distri-opensuse/blob/master/lib/opensusebasetest.pm#L1001 within the method "wait_boot_past_bootloader" that already checks for "emergency-shell" and "emergency-mode" and calls https://github.com/os-autoinst/os-autoinst-distri-opensuse/blob/master/lib/opensusebasetest.pm#L815 which calls https://github.com/os-autoinst/os-autoinst-distri-opensuse/blob/master/lib/utils.pm#L1050 which you extended in fe21fb3e2 already

Reply:
Currently I don't see a "smart" way to do this. I cannot use select_console as it is because there will be failure due to needles not matching. I don't want to duplicate code from root_console in wait_past_boot_loader. As it is a very specific type of failure, I will close this PR and when needed, we can collect the logs with some manual intervention.

Reply:
I still like your approach of handling that automatically. Collecting logs with manual intervention is ok but shouldn't be your job then but the task of the bug assignee. Both you and me work in "Quality Engineering". That means in the end we do not want to conduct manual testing ourselve anymore, right? ;)